lib/puppet/application/agent.rb

```diff
@@ -124,31 +124,36 @@ def help
 USAGE NOTES
 -----------
 'puppet agent' does its best to find a compromise between interactive
-use and daemon use. Run with no arguments and no configuration, it will
-go into the background, attempt to get a signed certificate, and retrieve
-and apply its configuration every 30 minutes.
+use and daemon use. If you run it with no arguments and no configuration, it
+goes into the background, attempts to get a signed certificate, and retrieves
+and applies its configuration every 30 minutes.
 
-Some flags are meant specifically for interactive use -- in particular,
+Some flags are meant specifically for interactive use --- in particular,
 'test', 'tags' and 'fingerprint' are useful.
 
 '--test' does a single run in the foreground with verbose logging, then exits.
-It will also exit if it can't get a valid catalog. The exit code after running
-with '--test' is 0 if the catalog was successfully applied, and 1 if the run
-either failed or wasn't attempted (due to another run already in progress).
+It also exits if it can't get a valid catalog. The exit code after running
+with '--test' is one of the following:
+
+* 0: The run succeeded with no changes or failures; the system was already in the desired state.
+* 1: The run failed, or wasn't attempted due to another run already in progress.
+* 2: The run succeeded, and some resources were changed.
+* 4: The run succeeded, and some resources failed.
+* 6: The run succeeded, and included both changes and failures.
```

Comment:
I suggested replacing the above with:

```
After running with '--test', an exit code of 0 (success) means the system was
already in the desired state, because the catalog was applied with no changes
or failures. See the '--detailed-exitcodes' option for more information.
```

This is better, because:
* It explains the mnemonic of the `--test` attribute: what is being "tested" is whether the system is already in the desired state.
* It avoids repeating information that is available elsewhere on the same page.
* It clearly links `--test` with `--detailed-exitcodes` in the first and most prominent place that `--test` is mentioned.

Reply:
@jbondpdx Ping?

Reply:
Hi @pillarsdotnet , thank you for your feedback! Because --test inherently includes the --detailed-exitcodes option, I think it's worth calling out in this usage section, rather than making users look for that information further down the page. I have added an explicit mention of the --detailed-exitcodes option because I agree with you that they should have been more clearly linked.